src/tests/linalg/test_linalg.f90

```diff
@@ -0,0 +1,25 @@
+program test_linalg
+  use stdlib_experimental_linalg, only: diag, eye
+  use stdlib_experimental_kinds, only: sp, dp, qp, &
+    int8, int16, int32, int64
+  implicit none
+
+  real(sp) :: a(5)
+  real(sp), allocatable :: d(:,:)
+  integer :: i, j
+
+  a = [1,2,3,4,5]
+  d = diag(a,0)
+
+  do i = 1, size(d,1)
+    write(*,*) (d(i,j),j=1,size(d,2))
+  end do
+  
+  a = diag(d,0)
+  print *, "a = ", a
+
+  print *, eye(5)
+
+  print *, all(eye(4) == diag([1,1,1,1]))
```

Comment:
Could the `check` function be used here?

Reply:
I will add more tests with calls to `check`. This was just temporary for development purposes.

Reply:
Nice. For internal stdlib tests, I suggest providing custom message in each `check` call so we can identify which failed, and also with `warn=.true.` to allow multiple tests to fail in a single test program.

Reply:
> and also with `warn=.true.` to allow multiple tests to fail in a single test program.

@milancurcic With `warn = .true.` multiple tests can fail, but `ctest` will report that all tests passed. It is ok when we develop it. But before merging into master, we should probably have `warn=.false.`.

Reply:
You're right, I forgot about that. We shouldn't do it then. It's important that ctest correctly reports failure.